modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RollUpIntegrationTest.java

```diff
@@ -0,0 +1,1144 @@
+				String bucket = result.getString("bucket_label");
+				userCount += result.getInt("user_count");
+				distinctUserCount += result.getInt("distinct_user_count");
+				atLeastOneRow = true;
+				if (RED.equals(bucket) || BLUE.equals(bucket)) {
+					// ok
+				} else {
+					fail("Bucket should be red or blue " + bucket);
+				}
+			}
+			if (!atLeastOneRow) {
+				fail("At least one row expected for query: "
+						+ queryGroupByBucketAndAction);
+			}
+			assertEquals(userCount, 2);
+			assertEquals(distinctUserCount, 2);
+		}
+	}
+
+	@Test(dependsOnMethods = { "t_PostImpressionTomorrowQA" })
+	public void t_CheckImpressionForTodayByBucketFromMySQLAllContext()
+			throws Exception {
+
+		String queryGroupByBucket = "select bucket_label, "
+				+ "count(user_id) as user_count, count(distinct user_id) as distinct_user_count  "
+				+ "from event_impression where hex(experiment_id) = '"
+				+ experiment.id.replace("-", "").toUpperCase()
+				+ "' and timestamp = '" + this.today
+				+ "' group by bucket_label";
+		try (Statement statement = connection.createStatement();
+				ResultSet result = statement.executeQuery(queryGroupByBucket)) {
+			if (result.next()) {
+				fail("should be no impressions for today " + queryGroupByBucket);
+			}
+		}
+	}
+
+	@Test(dependsOnMethods = { "t_PostImpressionTomorrowQA" })
+	public void t_CheckImpressionForTomorrowByBucketFromMySQLAllContext()
+			throws Exception {
+
+		String queryGroupByBucket = "select bucket_label, "
+				+ "count(user_id) as user_count, count(distinct user_id) as distinct_user_count  "
+				+ "from event_impression where hex(experiment_id) = '"
+				+ experiment.id.replace("-", "").toUpperCase()
+				+ "' and timestamp = '" + this.tomorrow
+				+ "' group by bucket_label";
+		try (Statement statement = connection.createStatement();
+				ResultSet result = statement.executeQuery(queryGroupByBucket)) {
+			int userCount = 0;
+			int distinctUserCount = 0;
+			boolean atLeastOneRow = false;
+			while (result.next()) {
+				String bucket = result.getString("bucket_label");
+				userCount += result.getInt("user_count");
+				distinctUserCount += result.getInt("distinct_user_count");
+				atLeastOneRow = true;
+			}
+			if (!atLeastOneRow) {
+				fail("At least one row expected for query: "
+						+ queryGroupByBucket);
+			}
+			assertEquals(userCount, 2);
+			assertEquals(distinctUserCount, 2);
+		}
+	}
+
+	@Test(dependsOnMethods = { "t_PostImpressionTomorrowQA" })
+	public void t_PostRollUpsForTomorrow() throws Exception {
+
+		AnalyticsParameters paramsQA = new AnalyticsParameters();
+		paramsQA.confidenceLevel = .9999999d;
+		paramsQA.actions = null;
+		paramsQA.context = QA;
+
+		long startTime = System.currentTimeMillis();
+		ExperimentCounts countsBeforeQA = postExperimentCounts(experiment, paramsQA);
+		long endTime = System.currentTimeMillis();
+
+		System.out.println("Before analytics time QA was: "
+				+ (endTime - startTime));
+
+		startTime = System.currentTimeMillis();
```

Comment:
1) Why do we need to measure the times here?
2) This one instance of creating the start time should be a little later, directly before postExperimentCounts is called, to make it more comparable. Also consider using [Instant or nanoTime](http://stackoverflow.com/a/3383047/3004221).